### DISCUSSION TITLE
iOS tableiView滚动 首次加载 cell 时会闪硕一次黑屏

### DISCUSSION THREAD

**Discussion by AppHock** [Tencent/libpag#2087] [Q&A] [answered]
pag版本：4.0.5.60

场景：tableViewCell展示封面图和pagView，默认展示封面图，隐藏 pagView，等到cell滚动到指定区域再隐藏封面图，展示 pagView 
在首次加载 cell 时，会出现一次黑屏的场景

背景：我在cellForRowAt 方法中设置了 pagView 的数据源，并调用了 flush，但是调用 flush 的时候，pagView是隐藏状态

思考：pagView 是支持再屏渲染吗？如果pagView 隐藏状态，调用 flush，好像就不会去做GPU绘制，也不会将最终的结果纹理渲染到 pagView 上，导致切换pagView 为显示时那一刻还是黑色的，调用了 play 才开始有画面

问题：为什么首次加载 cell时，pagView 会出现一次黑屏的问题

**Comment by kevingpqi123** [comment, marked as answer]
当 PAGView 不可见时，是不会触发渲染的，这里出现黑屏的原因可能是 pag 的首帧渲染相对耗时，没有控制好切换逻辑从而导致黑屏，提供几种解决思路：
（1）切换的过程中保持 PAGView 为可见状态，可以将透明度设置的足够低，首帧渲染完成的时候做切换；
（2）出现黑屏的情况是素材的复杂度相对较高，优化素材是最直接的方法，降低首帧渲染耗时；
（3）如果不涉及占位图替换和文本编辑，可以升级至 PAG 4.2 或 PAG 4.3 的最新版本，切换使用 PAGImageView，当渲染缓存成功后，后续就是读取缓存数据，无论是内存还是渲染速度都会提升很多
